### PR TITLE
Fix some minor warnings in the matching package

### PR DESCRIPTION
**Pull request by MichaelSnowden**
**What changed?**
I went through all the GoLand IDE inspection errors in the matching package, and I fixed the ones that seemed relevant. Most of them are little typos.

**Why?**
To clean things up a bit, make it easier to find actual errors from inspection, make sure these don't have to get included in other actual behavioral changes.

**How did you test it?**
I made sure not to include any behavioral changes, even small things like replacing `err ==` with `errors.Is`.

**Potential risks**

**Is hotfix candidate?**

### CODE REVIEW

File: service/matching/db.go [hunk 1]
```diff
@@ -296,17 +296,15 @@ func (db *taskQueueDB) CompleteTasksLessThan(
...
 func (db *taskQueueDB) DbStoresUserData() bool {
 	return db.taskQueue.OwnsUserData() && db.taskQueueKind == enumspb.TASK_QUEUE_KIND_NORMAL
 }
 
 // GetUserData returns the versioning data for this task queue. Do not mutate the returned pointer, as doing so
 // will cause cache inconsistency.
-func (db *taskQueueDB) GetUserData(
-	ctx context.Context,
-) (*persistencespb.VersionedTaskQueueUserData, chan struct{}, error) {
+func (db *taskQueueDB) GetUserData(context.Context) (*persistencespb.VersionedTaskQueueUserData, chan struct{}, error) {
```
Comment (yycptt) [inline service/matching/db.go:307]: nit: I thought the original version is better and makes the line shorter?

File: service/matching/task_reader.go [hunk 2]
```diff
@@ -235,10 +235,16 @@ func (tr *taskReader) getTaskBatchWithRange(
 	return response.Tasks, err
 }
 
+type tasksBatch struct {
+	Tasks           []*persistencespb.AllocatedTaskInfo
+	ReadLevel       int64
+	IsReadBatchDone bool
+}
```
Comment (yycptt) [inline service/matching/task_reader.go:242]: nit: getTasksBatchResponse maybe? read level and the done flag look like metadata to the returned batch of tasks.

do those fields have to be exported?

**Review by yycptt — APPROVED** [review]

File: service/matching/task_reader.go (hunk 2 above)
Comment (MichaelSnowden) [reply service/matching/task_reader.go]: Renamed and made them unexported

File: service/matching/db.go (hunk 1 above)
Comment (MichaelSnowden) [reply service/matching/db.go]: Reverted except removing the param name

**Pull request merged**